x-pack/plugins/apm/public/context/apm_service/use_agent_metadata_details_fetcher.ts

```diff
@@ -7,15 +7,19 @@
 
 import { useFetcher } from '../../hooks/use_fetcher';
 import { useUrlParams } from '../url_params_context/use_url_params';
+import { APIReturnType } from '../../services/rest/createCallApmApi';
 
-export function useServiceAgentNameFetcher(serviceName?: string) {
+export type AgentMetadataDetails = APIReturnType<'GET /api/apm/services/{serviceName}/metadata/details'>;
+
+export function useAgentMetadataDetailsFetcher(serviceName?: string) {
```

Comment:
This method is calling `'GET /api/apm/services/{serviceName}/metadata/details'`. Thus, shouldn't it be called

```suggestion
export function useServiceMetadataDetailsFetcher(serviceName?: string) {
```